### PR TITLE
atlas: use custom differ to detect unsynced first runs

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: atlas/resource_atlas_schema.go [hunk 1]
```diff
@@ -41,6 +43,61 @@ func newSchemaResource() *schema.Resource {
...
+					causes = append(causes, fmt.Sprintf("DROP COLUMN %q.%q", c.T.Name, d.C.Name))
+				}
+			}
+		}
+	}
+	if len(causes) > 0 {
+		return fmt.Errorf(`The database contains resources that Atlas wants to drop because they are not defined in the HCL file on the first run.
+- %s
+
+Please, follow the steps described in the project README:
```
Comment (rotemtam) [inline atlas/resource_atlas_schema.go:95]: To learn how to add an existing database to a project, read: "<link>

File: atlas/resource_atlas_schema.go [hunk 2]
```diff
@@ -41,6 +43,61 @@ func newSchemaResource() *schema.Resource {
...
 
+func customizeDiff(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	oldV, newV := diff.GetChange("hcl")
+	if oldV == nil {
+		return nil
+	}
+	s, ok := oldV.(string)
+	if !ok || s != "" {
+		return nil
+	}
```
Comment (masseelch) [inline atlas/resource_atlas_schema.go:54]: I think I am missing context here, why is this "okay"?

File: atlas/resource_atlas_schema.go (hunk 1 above)
Comment (rotemtam) [inline atlas/resource_atlas_schema.go:95]:
```suggestion
To learn how to add an existing database to a project, read: https://atlasgo.io/terraform-provider#working-with-an-existing-database
```

File: atlas/resource_atlas_schema.go (hunk 2 above)
Comment (a8m) [reply atlas/resource_atlas_schema.go:54]: See line 92. I want this check only on the first execution. i.e. when there is no previous state. Makes sense?

File: atlas/resource_atlas_schema.go [hunk 3]
```diff
@@ -41,6 +43,61 @@ func newSchemaResource() *schema.Resource {
...
+				}
+			}
+		}
+	}
+	if len(causes) > 0 {
+		return fmt.Errorf(`The database contains resources that Atlas wants to drop because they are not defined in the HCL file on the first run.
+- %s
+
+To learn how to add an existing database to a project, read: https://atlasgo.io/terraform-provider#working-with-an-existing-database
+https://github.com/ariga/terraform-provider-atlas#quick-start`, strings.Join(causes, "\n- "))
```
Comment (a8m) [inline atlas/resource_atlas_schema.go:96]:
```suggestion
To learn how to add an existing database to a project, read: https://atlasgo.io/terraform-provider#working-with-an-existing-database`, strings.Join(causes, "\n- "))
```

File: atlas/resource_atlas_schema.go [hunk 4]
```diff
@@ -41,6 +43,60 @@ func newSchemaResource() *schema.Resource {
...
+					causes = append(causes, fmt.Sprintf("DROP COLUMN %q.%q", c.T.Name, d.C.Name))
+				}
+			}
+		}
+	}
+	if len(causes) > 0 {
+		return fmt.Errorf(`The database contains resources that Atlas wants to drop because they are not defined in the HCL file on the first run.
+- %s
+
+To learn how to add an existing database to a project, read: https://atlasgo.io/terraform-provider#working-with-an-existing-database`, strings.Join(causes, "\n- "))
```
Comment (a8m) [inline atlas/resource_atlas_schema.go:95]:
```suggestion
To learn how to add an existing database to a project, read:
https://atlasgo.io/terraform-provider#working-with-an-existing-database`, strings.Join(causes, "\n- "))
```

**Review by rotemtam — APPROVED** [review]
🎉

**Pull request merged**